control/controlclient/auto.go

```diff
@@ -694,6 +694,17 @@ func (c *Auto) StartLogout() {
 func (c *Auto) Logout(ctx context.Context) error {
 	c.logf("client.Logout()")
 
+	c.mu.Lock()
+	loggedIn := c.loggedIn
+	c.mu.Unlock()
```

Comment:
if you combine this block with the one below you will only need to acquire the lock once. something like

```
c.mu.Lock()
if !c.loggedIn {
  c.mu.Unlock()
  return nil
}
...
c.mu.Unlock()
```

Reply:
👍🏼 nice, that's a lot cleaner